src/ftxui/dom/text.cpp

```diff
@@ -29,10 +29,15 @@ class Text : public Node {
     if (y > box_.y_max)
       return;
     for (wchar_t c : text_) {
-      if (x > box_.x_max)
-        return;
-      screen.at(x, y) = c;
-      x += wchar_width(c);
+      const int width = wchar_width(c);
+      if (width >= 1) {
+        if (x > box_.x_max)
+          return;
+        screen.PixelAt(x, y).character = c;
+      } else {
+        screen.PixelAt(x - 1, y).character += c;
```

Comment:
I was assuming users were using meaningful UTF8 and meaningful UTF8 doesn't starts with a codepoint of zero width.

But yes, this sounds like a good idea.

Reply:
No, my concern was whether accessing out-of-bounds locations would lead to a memory safety issue. However, I have looked at `Screen::PixelAt` and it appears to be safe, so there seems to be no such issue. Sorry for the false alarm.